rmw_fastrtps_shared_cpp/src/custom_publisher_info.cpp

```diff
@@ -160,6 +198,41 @@ void PubListener::set_on_new_event_callback(
   }
 }
 
+void
+PubListener::on_publication_matched(
+  eprosima::fastdds::dds::DataWriter * /* writer */,
+  const eprosima::fastdds::dds::PublicationMatchedStatus & info)
+{
+  rmw_event_type_t event_type = RMW_EVENT_INVALID;
+  {
+    std::lock_guard<std::mutex> lock(discovery_m_);
+    if (info.current_count_change == 1) {
```

Comment:
related to https://github.com/ros2/rmw_connextdds/pull/101#discussion_r1129753385, same question.

`current_count_change` will be either 1 or -1 always? otherwise, we will miss the event. as far as i see the specification, this could be 2 or -2 for example.

Reply:
These codes aren't changed. I only move it.  
The original code is located at   
https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_publisher_info.hpp#L70-L84

Reply:
I think on_publication_matched() was called while each DDS publication matched event happened. So the value of it will not exceed 1.

Reply:
Yes, it should be 1 or -1 always. Because the [listener](https://github.com/ros2/rmw_fastrtps/blob/8510fc1582383ad66092cf8b2236713df0e236e6/rmw_fastrtps_cpp/src/publisher.cpp#L267-L271) for `publication_matched`  is registered at the beginning.

https://github.com/eProsima/Fast-DDS/blob/bf181c8146f75741417590d6170e3395c2c86355/src/cpp/fastdds/publisher/DataWriterImpl.cpp#L1343-L1345 will be called every time from ([1](https://github.com/eProsima/Fast-DDS/blob/1535aef6276670fe215a3cfb35510966409497db/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp#L629), [2](https://github.com/eProsima/Fast-DDS/blob/1535aef6276670fe215a3cfb35510966409497db/src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp#L1195), ...).